### ISSUE TITLE
Editor.log files being held by Rider

### ISSUE THREAD

**Issue by vertxxyz** [JetBrains/resharper-unity#370]
Rider is preventing Unity from cycling the Editor.log file into Editor-prev.log by locking the Editor.log file.
This results in large, mostly unparsable, log files.

(To clarify these are the log files in AppData/Local/Unity/Editor)

**Comment by van800** [comment]
How do you came to the conslusion that Rider affects Unity logging behaviour?
To my understanding Unity starts new Editor.log when it is restarted. Is it true?

**Comment by vertxxyz** [comment]
After seeing the log grow in size on two machines, and not generate new ones or cycle to Editor-prev.log. Trying to delete the files on both machines resulted in Windows say Rider is locking the file (Even when Unity is closed.)
Closing Rider and running Unity successfully cycles the log file.
Coworker's machine without Rider logs just fine.

Seemed the likely cause based on that, but I didn't go any further to test after checking those few machines.

**Comment by van800** [comment]
Hm, thats weird, we don't have any functionality in Rider that can access Editor.log.
I will try to reproduce though.
What is your OS?
Have you done anything specific like opening Editor.log in Rider?

**Comment by vertxxyz** [comment]
Windows 10.
Huh, that is weird. I open those files in Sublime, so I see no reason for that to be the case.
And I am now on a 4th Win10 computer and I can replicate the issue, having Rider open during a restart of Unity keeps logging the same file, whereas when it's closed it cycles correctly.

**Comment by van800** [comment]
Doesn't reproduce from the first sign.
What Unity version do you have?
Could you please describe some steps to reproduce?

**Comment by vertxxyz** [comment]
2018.1.0b4 launched from the Unity Hub.
Load a script and wait for Rider to do all its things.
Close Unity.
Open Unity again (from Hub).
Note that Editor-prev.log remains the same and Editor.log grows as you repeat prior steps.

Close Unity.
Close Rider.
Open Unity.
Editor-prev.log has now been updated and a new Editor.log file is generated.

Here's a video if you'd like: https://youtu.be/G60MxKY11BE Unity crashes at the start because it's a healthy program 👍

**Comment by van800** [comment]
Thanks for the repro, I have reproduced it.

**Comment by van800** [comment]
![image](https://user-images.githubusercontent.com/1482681/36529521-d793d190-17b8-11e8-94d6-dedd1fe4ba6e.png)

**Comment by van800** [comment]
Found the solution, please update the following lines in Unity3dRider.cs
```
      proc.StartInfo.UseShellExecute = false;
      proc.StartInfo.WindowStyle = ProcessWindowStyle.Hidden;
      proc.StartInfo.CreateNoWindow = true;
      proc.StartInfo.RedirectStandardOutput = true;
```
into 
`      proc.StartInfo.UseShellExecute = false;`

The problem itself is about [Handle Inheritance](https://msdn.microsoft.com/en-us/library/windows/desktop/ms724466?f=255&MSPPError=-2147217396)

I will push fix to 2018.1. I expect it to get to EAP2.

**Comment by van800** [comment]
Available in the nighly builds from [toolbox](https://www.jetbrains.com/toolbox/download)
